src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,154 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * The Storage class is responsible for managing the storage and retrieval of tasks in the Duke chat-bot application.
+ * It handles saving tasks to a file and loading tasks from a file.
+ */
+public class Storage {
+    private static final String FILE_PATH = "./data/duke.txt";
+
+    /**
+     * Saves a list of tasks to a file.
+     *
+     * @param tasks The list of tasks to be saved.
+     */
+    public static void saveTasks(List<Task> tasks) {
+        try (FileWriter writer = new FileWriter(FILE_PATH)) {
+            for (Task task : tasks) {
+                writer.write(task.toFileString() + System.lineSeparator());
+            }
+        } catch (IOException e) {
+            System.err.println("Error saving tasks: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Loads tasks from a file and returns them as a list.
+     *
+     * @return A list of tasks loaded from the file.
+     * @throws DukeException If there is an issue loading tasks from the file.
+     */
+    public static List<Task> loadTasks() throws DukeException {
+        List<Task> tasks = new ArrayList<>();
+        createEmptyFile();
+        try (BufferedReader reader = new BufferedReader(new FileReader(FILE_PATH))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                Task task = createTaskFromLine(line);
+                if (task != null) {
+                    tasks.add(task);
+                }
+            }
+        } catch (IOException e) {
+            System.err.println("Error loading tasks: " + e.getMessage());
+        }
+
+        return tasks;
+    }
+
+    /**
+     * Creates an empty file and directory if they do not exist.
+     *
+     * @throws DukeException If there is an issue creating the file or directory.
+     */
+    private static void createEmptyFile() throws DukeException {
+        File directory = new File("./data");
+        File file = new File("./data/duke.txt");
+
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+
+        if (!file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                throw new DukeException("File Creation Failed");
+            }
+        }
+    }
+
+    /**
+     * Creates a Task object from a line of text read from the file.
+     *
+     * @param line The line of text representing a task.
+     * @return The Task object created from the line.
+     */
+    private static Task createTaskFromLine(String line) {
```

Comment:
Maybe can add more explanatory comments to clarify the purpose of each variables used and their roles would certainly enhance understanding, especially for uses who may not be familiar with the codebase yet.

Reply:
I have added detailed comments to do the method.